src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -159,9 +159,7 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
             // this is a no-op.  Emitting an error here, or when the original parameter was bound, would
             // adversely effect the compilation or potentially change overload resolution.  
             var compilation = this.DeclaringCompilation;
-            if (Type.ContainsDynamic() && 
-                compilation.HasDynamicEmitAttributes() &&
-                compilation.GetSpecialType(SpecialType.System_Boolean).SpecialType == SpecialType.System_Boolean)
```

Comment:
The link to a PR comment in the issue has the following information regarding the value returned by GetSpecialType:
if GetUseSiteDiagnostic() for it returns non-null DiagnosticInfo with Severity == error then it is bad/missing.